fenicsadapter/adapter_core.py

```diff
@@ -0,0 +1,194 @@
+"""
+This module consists of all core functionality of the FEniCS adapter .
+The module also consists of additional helper functions for the user
```

Comment:
Why does the module consists of two things? Would it be better to split the module in two parts? Example:
1. One module with the "core functionality"
2. One module with helper functions that can be used by the user and also the adapter implementation?